lib/include/openamp/virtio.h

```diff
@@ -503,6 +503,23 @@ static inline int virtio_reset_device(struct virtio_device *vdev)
 	return 0;
 }
 
+/**
+ * @brief Check if the virtio device support a specific feature.
+ *
+ * @param vdev		Pointer to device structure.
+ * @param feature_bit	Feature bit to check.
+ *
+ * @return true if the feature is supported, otherwise false.
+ */
+static inline bool virtio_has_feature(struct virtio_device *vdev,
+				      unsigned int feature_bit)
+{
+	if (!vdev)
+		return false;
+
+	return (vdev->features & (1UL << feature_bit)) != 0;
+}
```

Comment:
This function return  valid value only if virtio_negotiate_features has been called first . That also means that  it does not work for  virtio device role.

Reply:
Done, @arnopo Could you review again?

Reply:
> Yes, I consider assign back the final features to `vdev->features` when call `virtio_get_feature()` for VIRTIO_DEVICE side, so VIRTIO_DEVICE can use this `virtio_has_feature()` after called `virtio_get_feature()`. What do you think?

This does not prevent the use case where `virtio_has_feature` is called before `virtio_get_features`.
to be safe `virtio_has_feature` should call `virtio_get_features` if `vdev->features == NULL`